src/App.js

```diff
@@ -1,15 +1,45 @@
+import React from 'react';
+import { useState } from 'react';
 import './App.css';
 
 const App = () => {
 
+  const [count, setCount] = useState(0);
+  const [countertext, setCountertext] = useState('');
+
+  const addCounter = () => {
+    setCount(prevCount =>  prevCount  + 1)
+  }
+
+  const subtractCounter = () => {
+    setCount((prevCount) => prevCount - 1)
+  }
+
+
   return(
     <main>
         <h1>Counter App</h1>
-        <input type='text' class ="counter-text" placeholder='Enter name for your counter' />
-        <div class ="counter-count">{0}</div>
+        <input 
+          type='text' 
+          class ="counter-text" 
+          placeholder='Enter name for your counter' 
+          value={countertext} 
+          onChange={event => setCountertext(event.target.value)} 
```

Comment:
pass a function name to the onChange event instead of writing the function inside
makes the code cleaner to read and also better to update later on
i.e. write all functions before the return (like you have addCount and others)